content/blog/using-the-hpegl-hpe-greenlake-and-morpheus-terraform-providers-in-concert-to-create-provision-and-manage-hpe-pce-private-cloud-enterprise-vmaas-instances.md

```diff
@@ -0,0 +1,1492 @@
+---
+title: Using the hpegl and Morpheus Terraform Providers to provision a VMaaS workload
+date: 2025-01-15T14:58:20.593Z
+author: Eamonn O'Toole Senior Principal Cloud Developer, John Lenihan Software
+  Engineer, HPE Private Cloud
+authorimage: /img/head-shot.jpg
+disable: false
+tags:
+  - hpegl
+  - PCE
+  - Morpheus
+  - GreenLake
+  - HCL
+  - Terraform
+  - OpenTofu
+  - VMaaS
+---
+# **Introduction**
+
+Many customers of HPE’s Private Cloud Enterprise (PCE) offering want to use Terraform or the open-source alternative OpenTofu to create, provision and manage VMaaS (VM as a Service) VMs.  The cloud-based VMaaS service interacts with Morpheus (recently acquired by HPE) running on-premise.  There is a HPE GreenLake terraform provider called hpegl (HPE GreenLake) and a separate Morpheus provider.  Both providers complement each other and when used in concert offer a rich set of IaC capabilities.  In this blog post we will demonstrate how to use the two providers in concert to create, provision and manage VMaaS VM instances.  Central to this is a hpegl data-source which can be used to retrieve an Access Token and URL for the on-premise Morpheus instance which are then passed to the Morpheus provider.
+
+# **IaC and PCE**
+
+IaC is now a well-established approach to configuring, creating and managing resources.  Terraform and more recently its OpenSource alternative OpenTofu along with per-service providers are especially popular.  For HPE’s PCE (Private Cloud Enterprise) VMaaS (VM as a Service) service there are two relevant providers: 
+
+* hpegl (HPE GreenLake), which interacts with the PCE Cloud Services (IAM and VMaaS) 
```

Comment:
suggestion:
hpegl which interacts with the HPE GreenLake for Private Cloud Enterprise cloud services such as VMaaS and Identity and Access Management (IAM).

I suggest to spell out IAM (Identify and Access Management).

Reply:
Done.